java/flight/flight-sql/src/main/java/org/apache/arrow/flight/sql/FlightSqlClient.java

```diff
@@ -1048,14 +1049,34 @@ private Schema deserializeSchema(final ByteString bytes) {
     public FlightInfo execute(final CallOption... options) {
       checkOpen();
 
-      final FlightDescriptor descriptor = FlightDescriptor
+      FlightDescriptor descriptor = FlightDescriptor
           .command(Any.pack(CommandPreparedStatementQuery.newBuilder()
                   .setPreparedStatementHandle(preparedStatementResult.getPreparedStatementHandle())
                   .build())
               .toByteArray());
 
       if (parameterBindingRoot != null && parameterBindingRoot.getRowCount() > 0) {
-        putParameters(descriptor, options);
+        SyncPutListener putListener = putParameters(descriptor, options);
+
+        try {
+          final PutResult read = putListener.read();
```

Comment:
Is it necessary to do this even in the case where no parameters were returned?

Reply:
I don't think it is necessary but I guess it makes the response more predictable.

Reply:
I was thinking we shouldn't do parameter binding (line 1059) if we already got the parameter schema and it was empty.

Reply:
Ok, I can check for this.

Reply:
Done